Src/Microsoft.Dynamic/Hosting/Shell/OptionsParser.cs

```diff
@@ -327,7 +333,8 @@ public class OptionsParser<TConsoleOptions> : OptionsParser
                 { "-X:PrivateBinding",           "Enable binding to private members" },
                 { "-X:ShowClrExceptions",        "Display CLS Exception information" },
                 { "-X:TabCompletion",            "Enable TabCompletion mode" },
-                { "-X:ColorfulConsole",          "Enable ColorfulConsole" },
+                { "-X:ColorfulConsole[=(dark|light)]",
```

Comment:
Does using `=` instead of a space work in both 2 and 3? I know I changed 3 to more closely resemble CPython but I forget the details (and am unable to test right now). Maybe not a big deal since I don't plan on another release of 2...

Reply:
It cannot be a space iso `=` here because the argument is optional. With a space it would be mistaken for a file argument.

This will not work with 2 (which BTW is not buildable right now because the DLR is on net452). Also, `-X:ColorfulConsole=light` (with `:` iso space) does not work with 3. This by itself is not a problem (since there is no backward compatibility need here nor is this form shown in the help printout), but it kind of makes the DLR not self-consistent and unable to process the optional argument at all. It happens to work with 3 because of all the extra tweaks that it does to options before handing them over to the base class in the DLR for processing.

I can add support for this form to the DLR, just for options with an optional argument (of which this is the first one). An alternative could be to modify the DLR that any mandatory option argument can be given separated by a space ***or*** `=`. The DLR has between 1 and 3 of those (depending on a feature set). What do you think?

Reply:
I'm having a hard time making up my mind because I don't know who (if anyone) outside of us is using the cli parsing. If we allow both spaces and equal on mandatory arguments I think we'd be inconsistent with Python? Probably easiest to add support for this form. Could even just be hard coded in the switch statement (e.g. `-X:ColorfulConsole=true`)...